cluster/schema/meta_class.go

```diff
@@ -478,6 +480,20 @@ func (m *metaClass) applyShardProcess(name string, action command.TenantProcessR
 
 		if count == len(processes) {
 			copy.Status = req.Tenant.Status
+			// Whenever a tenant is frozen, we send a notification the querier nodes registered
+			// with the querier manager. This call to NotifyClassTenantDataUpdate is non-blocking
+			// and will skip/return an error for any querier nodes which have a full notification
+			// channel.
+			err := m.querierManager.NotifyClassTenantDataUpdate(
+				querier.ClassTenant{
+					ClassName:  m.Class.Class,
+					TenantName: name,
+				},
+			)
+			if err != nil {
+				// TODO log
+				fmt.Println("failed to notify some queriers", err)
+			}
```

Comment:
@moogacs do you think this is ok to add here or should we put the call behind a config flag just in. case? The idea is that it does "nothing" if there are no querier nodes registered.

Reply:
Let's put it behind a config flag. The apply loop is very critical and it's better to avoid any expensive operation here. 

Could that loop just queue a message in a channel ? (with a failsafe that if the channel is full/blocking we just ignore it, log a warn and move on so that we don't bloc k?)

Then we can have one or multiple reader on that channel that will ensure the notify will be send to the queriers ? Idea being to cut out of the apply loop any code that might do a network call/hang/etc...

Reply:
> Could that operation instead just queue a message in a channel ? (with a failsafe that if the channel is full/blocking we just ignore it, log a warn and move on so that we don't bloc k?)

That is what NotifyClassTenantDataUpdate does:
```
func (qm *QuerierManager) NotifyClassTenantDataUpdate(ct ClassTenant) error {
        ....
	for q := range qm.registeredQueriers {
		select {
		case q.classTenantDataUpdates <- ct:
                        ...
		default:
                        ...
			continue
		}
	}
        ...
}
```

> Maybe we could move up the code to instead be at the raft/apply level directly and avoid the intricacies of querier notification at the schema level ? IMO that package should only care about the schema and that's it.

> Happy to discuss what is the best place to plus the notification from the schema flow (or if this location is the current best that's a-ok too!)

I'll look into these and potentially reach out with questions, thanks

Reply:
I introduced a channel layer of indirection which is only used when the metadata server is enabled. However, the only ways I could think to move the location out of the metaclass seemed overcomplicated. Let me know if you want to discuss more. I'd prefer to merge this as is given the guardrails I have in place, and revisit later.

Reply:
Discussed with @reyreaud-l , given the current code structure we decided to leave this channel in meta_class.go for now and add some testing to ensure it doesn't error or block the raft apply path. Refactoring this out is probably possible, but seems like more work/risk than is worth at this time. If we stick with using the channel, we may want to revisit this in the future